packages/auto-tls/src/auto-tls.ts

```diff
@@ -0,0 +1,200 @@
+import { ClientAuth } from '@libp2p/http-fetch/auth'
+import { serviceDependencies, stop } from '@libp2p/interface'
+import { debounce } from '@libp2p/utils/debounce'
+import { isLoopback } from '@libp2p/utils/multiaddr/is-loopback'
+import { isPrivate } from '@libp2p/utils/multiaddr/is-private'
+import { QUICV1, TCP, WebSockets, WebSocketsSecure, WebTransport } from '@multiformats/multiaddr-matcher'
+import { Crypto } from '@peculiar/webcrypto'
+import * as x509 from '@peculiar/x509'
+import * as acmeClient from 'acme-client'
+import { base36 } from 'multiformats/bases/base36'
+import type { AutoTLSComponents, AutoTLSInit, AutoTLS as AutoTLSInterface } from './index.js'
+import type { PeerId, PrivateKey, Logger, TypedEventTarget, Libp2pEvents, AbortOptions, TLSCertificate } from '@libp2p/interface'
+import type { AddressManager } from '@libp2p/interface-internal'
+import type { DebouncedFunction } from '@libp2p/utils/debounce'
+import type { Multiaddr } from '@multiformats/multiaddr'
+
+const crypto = new Crypto()
+x509.cryptoProvider.set(crypto)
+
+type CertificateEvent = 'certificate:provision' | 'certificate:renew'
```

Comment:
```suggestion
type CertificateEvent = 'certificate:provision' | 'certificate:renew' | `certificate:load`
```

add event for loading a cert during startup?

Reply:
I thought about doing something like this but it wasn't clear to me what the actual difference between `certificate:provision` and `certificate:load` would be, they both just mean "a certificate is available and one wasn't before", whereas `certificate:renew` means "a certificate is available but you may need to undo what you did before to start using it".